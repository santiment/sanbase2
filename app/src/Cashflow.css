.cashflow-head {
  margin-bottom: 1em;
}

.cashflow-head p {
  text-align: right;
}
