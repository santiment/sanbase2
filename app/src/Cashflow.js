import React from 'react'
import ReactTable from 'react-table'
import { connect } from 'react-redux'
import {
  compose,
  pure,
  lifecycle
} from 'recompose'
import 'react-table/react-table.css'
import moment from 'moment'
import { formatNumber } from './utils/formatting'

const formatDate = date => moment(date).format('YYYY-MM-DD')

const formatLastOutgoingWallet = wallets => {
  return wallets.map((wallet, index) => {
    const lastOutgoing = wallet.last_outgoing !== null
      ? formatDate(wallet.last_outgoing) : 'No recent transfers'
    return (
      <div key={index}>
        { lastOutgoing }
      </div>
    )
  })
}

const formatTxOutWallet = wallets => {
  return wallets.map((wallet, index) => {
    const txOut = wallet.tx_out || 0
    return (
      <div key={index}>
        {txOut.toLocaleString('en-US')}
      </div>
    )
  })
}

const formatBalanceWallet = ({wallets, ethPrice}) => {
  return wallets.map((wallet, index) => {
    const balance = wallet.balance || 0
    return (
      <div className='wallet' key={index}>
        <div className='usd first'>{formatNumber((balance * ethPrice), 'USD')}</div>
        <div className='eth'>
          <a className='address' href={'https://etherscan.io/address/' + wallet.address} target='_blank'>Îž{ balance.toLocaleString('en-US') }
            <i className='fa fa-external-link' />
          </a>
        </div>
      </div>
    )
  })
}

const formatMarketCapProject = cap => {
  if (cap !== null) {
    return formatNumber(cap, 'USD')
  } else {
    return 'No data'
  }
}

const columns = [{
  Header: 'Project',
  id: 'project',
  filterable: true,
  sortable: true,
  accessor: d => ({
    name: d.name,
    ticker: d.ticker
  }),
  Cell: ({value}) => <div>{value.name} ({value.ticker})</div>,
  filterMethod: (filter, row) => {
    return row[filter.id].name.toLowerCase().indexOf(filter.value) !== -1 ||
      row[filter.id].ticker.toLowerCase().indexOf(filter.value) !== -1
  }
}, {
  Header: 'Market Cap',
  id: 'market_cap_usd',
  sortable: true,
  accessor: 'market_cap_usd',
  Cell: props => <span>{formatMarketCapProject(props.value)}</span>, // Custom cell components!
  sortMethod: (a, b) => {
    const _a = parseInt(a, 10)
    const _b = parseInt(b, 10)
    if (_a === _b) {
      return 0
    }
    return _a > _b ? 1 : -1
  }
}, {
  Header: 'Balance (USD/ETH)',
  id: 'balance',
  accessor: d => ({
    ethPrice: d.ethPrice,
    wallets: d.wallets
  }),
  Cell: props => <div>{formatBalanceWallet(props.value)}</div>
}, {
  Header: 'Last outgoing TX',
  id: 'tx',
  accessor: d => d.wallets,
  Cell: props => <div>{formatLastOutgoingWallet(props.value)}</div>
}, {
  Header: 'ETH sent',
  id: 'sent',
  accessor: d => d.wallets,
  Cell: props => <div>{formatTxOutWallet(props.value)}</div>
}]

export const Cashflow = ({
  projects,
  loading
}) => (
  <div>
    <h1>Cashflow</h1>
    <ReactTable
      loading={loading}
      showPagination={false}
      showPaginationTop={false}
      showPaginationBottom={false}
      sortable={false}
      resizable
      defaultSorted={[
        {
          id: 'market_cap_usd',
          desc: true
        }
      ]}
      className='-striped -highlight'
      data={projects}
      columns={columns}
    />
  </div>
)

const mapStateToProps = state => {
  return {
    projects: state.projects.items,
    loading: state.projects.loading
  }
}

const mapDispatchToProps = dispatch => {
  return {
    retrieveProjects: () => {
      dispatch({
        types: ['LOADING_PROJECTS', 'SUCCESS_PROJECTS', 'FAILED_PROJECTS'],
        payload: {
          client: 'sanbaseClient',
          request: {
            url: `/cashflow`
          }
        }
      })
    }
  }
}

const enhance = compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  lifecycle({
    componentDidMount () {
      this.props.retrieveProjects()
    }
  }),
  pure
)

export default enhance(Cashflow)
