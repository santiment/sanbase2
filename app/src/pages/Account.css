.account-control {
  display: flex;
  justify-content: space-between;
}

@media (min-width: 1024px) {
  .account {
    width: 720px;
  }

  .account .panel form {
    width: 40%;
  }

  .account-control {
    display: block;
  }
}
