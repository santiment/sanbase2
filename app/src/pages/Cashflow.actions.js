export const retrieveProjects = {
  types: ['LOADING_PROJECTS', 'SUCCESS_PROJECTS', 'FAILED_PROJECTS'],
  payload: {
    client: 'sanbaseClient',
    request: {
      url: `/cashflow`
    }
  }
}
