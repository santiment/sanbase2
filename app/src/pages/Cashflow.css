.cashflow {
  width: 80%;
  max-width: 80%;
  min-width: 80%;
}

.cashflow-head {
  display: flex;
  flex-direction: column;
  margin-top: 1em;
  margin-bottom: 1em;
  padding: 0 16px;
}

.cashflow-title {
  display: flex;
  align-items: center;
}

.cashflow-title span {
  border-left: 1px solid #3d4450;
  margin: 18px 0 0 16px;
  padding-left: 16px;
}

.cashflow-head h1 {
  font-family: 'Source Sans Pro', sans-serif;
  margin: 0;
  color: rgba(0, 0, 0, 0.7) !important;
}

.cashflow-head p {
  text-align: right;
  font-size: 12px;
  margin-top: 21px;
}

.cashflow-indev-message {
  margin-bottom: 16px;
}

.row {
  display: flex;
  width: 100%;
  justify-content: space-between;
  padding: 0 16px 5px 16px;
}

.datatables-filter {
  margin-bottom: 16px;
  min-width: 25%;
}

.datatables-filter label {
  width: 100%;
  margin-bottom: 0.5rem;
  display: inline-block;
}

.datatables-filter input {
  width: 100%;
  margin-left: 0;
  padding: 8px;
  line-height: 1.25;
  color: #464a4c;
  border: 1px solid rgba(0, 0, 0, 0.15);
  background-clip: padding-box;
  outline-offset: -2px;
  overflow: visible;
  font-size: 0.875rem;
  border-radius: 0.2rem;
}

.datatables-filter input:focus {
  color: #464a4c;
  background-color: #fff;
  border-color: #5cb3fd;
  outline: 0;
}

.rt-thead.-filters {
  display: none;
}

.datatables-info label {
  font-size: 12px;
  margin-bottom: 16px;
  display: inline-block;
}

.cashflow .ReactTable {
  border-top: none;
  border-left: none;
  border-right: none;
  margin-left: -16px;
  margin-right: -16px;
  overflow-y: hidden;
  min-height: 350px;
}

.ReactTable .rt-tbody .rt-tr-group {
  border-bottom: solid 1px rgba(0, 0, 0, 0.14);
}

.rt-td:hover {
  cursor: pointer;
}

.ReactTable .rt-tbody .rt-td {
  border-right: 1px solid rgba(0, 0, 0, 0.05);
  padding: 8px;
}

.rt-tr:hover {
  background-color: #f2f2f2 !important;
  transition: background 0.25s ease-in;
}

.ReactTable .rt-th {
  background-color: white;
  border: none !important;
  padding: 10px !important;
  font-size: 12px;
  outline: none;
  text-transform: uppercase;
  font-family: 'Nunito Sans', sans-serif;
  letter-spacing: 0.03em;
  white-space: normal;
}

.ReactTable .rt-td > div {
  font-size: 18px;
  line-height: 1.2;
}

.ReactTable .rt-thead .rt-tr {
  text-align: right;
}

.ReactTable .rt-thead .rt-tr .rt-th:nth-child(2) {
  text-align: left;
}

.ReactTable .rt-thead .rt-tr .rt-th:nth-child(7),
.ReactTable .rt-thead .rt-tr .rt-th:last-child,
.ReactTable .rt-tbody .rt-td:last-child > div {
  text-align: center;
}

.ticker {
  font-size: 14px;
}

.-sort-desc,
.-sort-asc {
  background-color: rgba(0, 150, 99, 0.08);
}

.rt-td .overview-name {
  border-bottom: 1px solid #b1b7c1;
  display: inline-block;
  padding-bottom: 1px;
  padding-top: 12px;
  text-align: left;
  overflow: hidden;
  text-overflow: ellipsis;
}

.rt-td .overview-name:hover {
  color: #26a987;
}

.rt-td .overview-price,
.rt-td .overview-volume,
.rt-td .overview-marketcap,
.rt-td .overview-balance,
.rt-td .overview-ethspent,
.rt-td .overview-devactivity,
.rt-td .overview-activeaddresses {
  padding-top: 3px;
  text-align: right;
}

.rt-td .overview-pb {
  padding-top: 12px;
  text-align: center;
}

.rt-td .overview-marketcap,
.rt-td .overview-ethspent,
.rt-td .overview-devactivity,
.rt-td .overview-activeaddresses {
  padding-top: 12px;
}

.rt-td .overview-ticker {
  font-size: 14px;
  line-height: 1 !important;
  padding-top: 6px;
  text-align: center;
}

.ReactTable .wallet {
  display: flex;
  flex-direction: column;
}

.ReactTable .wallet .usd {
  flex: 1;
}

.ReactTable .wallet .eth {
  font-size: 14px;
}

.ReactTable .wallet .eth i {
  font-size: 15px;
}

.findwallet {
  border-bottom: none;
}

.ReactTable .cell-signals {
  width: 100%;
  height: 100%;
  padding: 15px 0 0 0;
  text-align: center;
}

.eth-sent-item {
  text-align: right;
}

@media all and (min-width: 768px) and (max-width: 1024px) {
  .rt-td > div {
    font-size: 16px;
    padding: 0;
  }

  .rt-td .overview-ticker {
    font-size: 12px;
    padding-top: 4px;
  }

  .rt-td .overview-name,
  .rt-td .overview-marketcap,
  .rt-td .overview-ethspent,
  .rt-td .overview-devactivity,
  .rt-td .overview-activeaddresses {
    padding-top: 9px;
  }
}

@media all and (min-width: 768px) and (max-width: 1440px) {
  .cashflow {
    width: 100%;
    min-width: 100%;
  }
}
