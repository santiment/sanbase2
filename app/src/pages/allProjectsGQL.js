import gql from 'graphql-tag'

const allProjectsGQL = gql`{
  allProjects {
    name
    rank
    description
    ticker
    marketSegment
    priceUsd
    percentChange24h
    volumeUsd
    volumeChange24h
    ethSpent
    averageDevActivity
    marketcapUsd
    ethBalance
    btcBalance
    usdBalance
    ethAddresses {
      address
    }
    twitterData {
      followersCount
    }
    signals {
      name
      description
    }
  }
}`

export default allProjectsGQL
