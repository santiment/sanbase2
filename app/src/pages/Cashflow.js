import React from 'react'
import ReactTable from 'react-table'
import { connect } from 'react-redux'
import {
  compose,
  pure,
  lifecycle
} from 'recompose'
import 'react-table/react-table.css'
import moment from 'moment'
import { formatNumber } from '../utils/formatting'
import ProjectIcon from './../components/ProjectIcon'
import {
  sortDate,
  sortBalances,
  sortTxOut,
  simpleSort
} from './../utils/sortMethods'
import { retrieveProjects } from './Cashflow.actions.js'
import './Cashflow.css'

const formatDate = date => moment(date).format('YYYY-MM-DD')

const formatLastOutgoingWallet = wallets => {
  return wallets.map((wallet, index) => {
    const lastOutgoing = wallet.last_outgoing !== null
      ? formatDate(wallet.last_outgoing) : 'No recent transfers'
    return (
      <div key={index}>
        { lastOutgoing }
      </div>
    )
  })
}

const formatTxOutWallet = wallets => {
  return wallets.map((wallet, index) => {
    const txOut = wallet.tx_out || '0.00'
    return (
      <div key={index}>
        {txOut.toLocaleString('en-US')}
      </div>
    )
  })
}

const formatBalanceWallet = ({wallets, ethPrice}) => {
  return wallets.map((wallet, index) => {
    const balance = wallet.balance || 0
    return (
      <div className='wallet' key={index}>
        <div className='usd first'>{formatNumber((balance * ethPrice), 'USD')}</div>
        <div className='eth'>
          <a
            className='address'
            href={'https://etherscan.io/address/' + wallet.address}
            target='_blank'>Îž{formatNumber(balance)}&nbsp;
            <i className='fa fa-external-link' />
          </a>
        </div>
      </div>
    )
  })
}

const formatMarketCapProject = cap => {
  if (cap !== null) {
    return formatNumber(cap, 'USD')
  } else {
    return 'No data'
  }
}

const getFilter = search => {
  if (search) {
    return [{
      id: 'project',
      value: search
    }]
  }
  return []
}

const columns = [{
  Header: 'Project',
  id: 'project',
  filterable: true,
  sortable: true,
  width: 350,
  accessor: d => ({
    name: d.name,
    ticker: d.ticker
  }),
  Cell: ({value}) => (
    <div>
      <ProjectIcon name={value.name} /> {value.name} ({value.ticker})
    </div>
  ),
  filterMethod: (filter, row) => {
    const name = row[filter.id].name || ''
    const ticker = row[filter.id].ticker || ''
    return name.toLowerCase().indexOf(filter.value) !== -1 ||
      ticker.toLowerCase().indexOf(filter.value) !== -1
  }
}, {
  Header: 'Market Cap',
  id: 'market_cap_usd',
  minWidth: 150,
  accessor: 'market_cap_usd',
  Cell: ({value}) => <div className='market-cap'>{formatMarketCapProject(value)}</div>,
  sortable: true,
  sortMethod: (a, b) => simpleSort(parseInt(a, 10), parseInt(b, 10))
}, {
  Header: 'Balance (USD/ETH)',
  id: 'balance',
  minWidth: 250,
  accessor: d => ({
    ethPrice: d.ethPrice,
    wallets: d.wallets
  }),
  Cell: ({value}) => <div>{formatBalanceWallet(value)}</div>,
  sortable: true,
  sortMethod: (a, b) => sortBalances(a, b)
}, {
  Header: 'Last outgoing TX',
  id: 'tx',
  accessor: d => d.wallets,
  Cell: ({value}) => <div>{formatLastOutgoingWallet(value)}</div>,
  sortable: true,
  sortMethod: (a, b, isDesc) => (
    sortDate(a[0].last_outgoing, b[0].last_outgoing, isDesc)
  )
}, {
  Header: 'ETH sent',
  id: 'sent',
  accessor: d => d.wallets,
  Cell: ({value}) => <div className='eth-sent-item'>{formatTxOutWallet(value)}</div>,
  sortable: true,
  sortMethod: (a, b) => sortTxOut(a, b)
}]

export const Cashflow = ({
  projects,
  loading,
  onSearch,
  search,
  tableInfo
}) => (
  <div className='page cashflow'>
    <div className='cashflow-head'>
      <h1>Cash Flow</h1>
      <p>
        brought to you by <a
          href='https://santiment.net'
          rel='noopener noreferrer'
          target='_blank'>Santiment</a>
        <br />
        NOTE: This app is a prototype.
        We give no guarantee data is correct as we are in active development.
      </p>
    </div>
    <div className='panel'>
      <div className='row'>
        <div className='datatables-info'>
          <label>
            Showing {
              (tableInfo.visibleItems !== 0)
                ? (tableInfo.page - 1) * tableInfo.pageSize + 1
                : 0
            } to {
              tableInfo.page * tableInfo.pageSize
            } of {tableInfo.visibleItems}
            &nbsp;entries&nbsp;
            {tableInfo.visibleItems !== projects.length &&
              `(filtered from ${projects.length} total entries)`}
          </label>
        </div>
        <div className='datatables-filter'>
          <label>
            <input placeholder='Search' onKeyUp={onSearch} />
          </label>
        </div>
      </div>
      <ReactTable
        loading={loading}
        showPagination={false}
        showPaginationTop={false}
        showPaginationBottom={false}
        defaultPageSize={projects.items ? projects.items.length : 32}
        sortable={false}
        resizable
        defaultSorted={[
          {
            id: 'market_cap_usd',
            desc: false
          }
        ]}
        className='-highlight'
        data={projects}
        columns={columns}
        filtered={getFilter(search)}
      />
    </div>
  </div>
)

const mapStateToProps = state => {
  return {
    projects: state.projects.items,
    loading: state.projects.loading,
    search: state.projects.search,
    tableInfo: state.projects.tableInfo
  }
}

const mapDispatchToProps = dispatch => {
  return {
    retrieveProjects: () => dispatch(retrieveProjects),
    onSearch: (event) => {
      dispatch({
        type: 'SET_SEARCH',
        payload: {
          search: event.target.value.toLowerCase()
        }
      })
    }
  }
}

const enhance = compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  lifecycle({
    componentDidMount () {
      this.props.retrieveProjects()
    }
  }),
  pure
)

export default enhance(Cashflow)
