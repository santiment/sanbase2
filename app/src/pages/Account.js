import React from 'react'
import { connect } from 'react-redux'
import { Helmet } from 'react-helmet'
import Raven from 'raven-js'
import { Redirect } from 'react-router-dom'
import { graphql } from 'react-apollo'
import gql from 'graphql-tag'
import {
  compose,
  withState,
  pure
} from 'recompose'
import {
  Form,
  Input,
  Message,
  Divider,
  Button
} from 'semantic-ui-react'
import { FadeIn } from 'animate-components'
import copy from 'copy-to-clipboard'
import Balance from './../components/Balance'
import { EmailField } from './../pages/Login/EmailLogin'
import { Form as ReactForm } from 'react-form'
import './Account.css'
const validate = require('validate.js')

const validateFields = email => {
  var constraints = {
    email: {
      email: true
    }
  }
  return validate({email}, constraints)
}

const errorValidator = ({email}) => {
  const validation = validateFields(email)
  return {
    email: validation && validation.email
  }
}

const successValidator = ({email}) => {
  const validation = validateFields(email)
  return {
    email: typeof validation === 'undefined' || !validation.email
  }
}

export const Account = ({
  user,
  loading,
  logout,
  changeEmail,
  changedEmail,
  isError = false,
  isSuccess = false,
  isPending,
  onSuccess,
  onError,
  onPending
}) => {
  if (user && !user.username) {
    return (
      <Redirect to={{
        pathname: '/'
      }} />
    )
  }
  return (
    <div className='page account'>
      <Helmet>
        <title>
          SANbase: Settings
        </title>
      </Helmet>
      <div className='page-head'>
        <h1>Account settings</h1>
      </div>
      {!user.email &&
        <Message
          className='account-message'
          warning
          header='Email is not added yet!'
          list={[
            'For acces your dashboard from mobile device, you should add email address.'
          ]}
        />}
      {isSuccess &&
        <Message
          className='account-message'
          positive
          content={`Email was changed to ${user.email}!`}
        />}
      {isError &&
        <Message
          className='account-message'
          negative
          header='Email is not changed!'
          list={[
            'Try again later...'
          ]}
        />}
      <div className='panel'>
        <ReactForm
          validateError={errorValidator}
          validateSuccess={successValidator}
          onSubmitFailure={(error, ...rest) => {
            onError(true)
            Raven.captureException(`User try to change email: ${error} ${rest}`)
          }}
          onSubmit={(values, _, formApi) => {
            onPending(true)
            changeEmail({variables: {...values}})
            .then(data => {
              onPending(false)
              onSuccess(true)
              onError(false)
              changedEmail(values.email)
              formApi.resetAll()
            })
            .catch(error => {
              onPending(false)
              onError(true)
              Raven.captureException(`User try to change email: ${error}`)
            })
          }}>
          {formApi => (
            <form
              className='account-settings-email'
              onSubmit={formApi.submitForm}
              autoComplete='off'>
              <EmailField
                autoFocus={false}
                disabled={isPending}
                placeholder={user.email || undefined}
                className='account-settings-email__input'
                formApi={formApi} />

              {formApi.getSuccess().email &&
              <FadeIn
                className='account-settings-email__button-container'
                duration='0.7s' timingFunction='ease-in' as='div'>
                <Button
                  disabled={
                    !formApi.getSuccess().email ||
                    isPending
                  }
                  positive={!!formApi.getSuccess().email}
                  type='submit'>
                  {isPending ? 'Waiting...' : 'Submit'}
                </Button>
              </FadeIn>}
            </form>
          )}
        </ReactForm>
        <br />
        <Form loading={loading}>
          <Form.Field>
            <label>Username or Eth Public Key</label>
            <Input
              input={{readOnly: true}}
              action={{
                color: 'teal',
                labelPosition: 'right',
                icon: 'copy',
                content: 'Copy',
                onClick: () => copy(user.username)
              }}
              defaultValue={user.username}
            />
          </Form.Field>
        </Form>
        <h3>Wallets</h3>
        <Divider />
        <Balance user={user} />
        <h3>Sessions</h3>
        <Divider />
        <div className='account-control'>
          <p>Your current session</p>
          <Button
            basic
            color='red'
            onClick={logout}
          >Log out</Button>
        </div>
      </div>
    </div>
  )
}

const mapStateToProps = state => {
  return {
    user: state.user.data,
    loading: state.user.isLoading
  }
}

const mapDispatchToProps = dispatch => {
  return {
    logout: () => {
      dispatch({
        type: 'SUCCESS_LOGOUT'
      })
    },
    changedEmail: email => {
      dispatch({
        type: 'CHANGE_EMAIL',
        email
      })
    }
  }
}

const changeEmailGQL = gql`
  mutation changeEmail($email: String!) {
    changeEmail(email: $email) {
      email
    }
  }
`

const enhance = compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  withState('isPending', 'onPending', false),
  withState('isError', 'onError', false),
  withState('isSuccess', 'onSuccess', false),
  graphql(changeEmailGQL, {
    name: 'changeEmail'
  }),
  pure
)

export default enhance(Account)
