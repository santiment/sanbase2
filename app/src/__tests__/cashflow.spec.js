/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Cashflow } from './../pages/Cashflow'

const projects = [
  {
    balance: '41994.09',
    id: 12,
    logo_url: 'aeternity.png',
    market_cap_usd: '133759577.0',
    name: 'Aeternity',
    ticker: 'AE',
    wallets: [
      {
        address: '0x15c19E6c203E2c34D1EDFb26626bfc4F65eF96F0',
        balance: '41994.09',
        last_outgoing: null,
        tx_out: null
      }
    ]
  },
  {
    balance: '264617.29',
    id: 15,
    logo_url: 'aragon.png',
    market_cap_usd: '66649211.0',
    name: 'Aragon',
    ticker: 'ANT',
    wallets: [
      {
        address: '0xcafE1A77e84698c83CA8931F54A755176eF75f2C',
        balance: '264617.29',
        last_outgoing: null,
        tx_out: null
      }
    ]
  }
]

const marketSegments = {
  advertising: 'Advertising',
  blockchain_network: 'Blockchain Network',
  data: 'Data',
  digital_identity: 'Digital Identity',
  financial: 'Financial',
  gambling: 'Gambling',
  gaming: 'Gaming',
  legal: 'Legal',
  media: 'Media',
  prediction_market: 'Prediction Market',
  protocol: 'Protocol',
  transportation: 'Transportation',
  unknown: null
}
const categories = {}

describe('Cashflow container', () => {
  it('it should render correctly', () => {
    const login = shallow(
      <Cashflow
        projects={projects}
        tableInfo={{
          visibleItems: 32,
          pageSize: 32,
          page: 1
        }}
        marketSegments={marketSegments}
        categories={categories}
        match={{ path: '/products' }}
      />
    )
    expect(toJson(login)).toMatchSnapshot()
  })
})
