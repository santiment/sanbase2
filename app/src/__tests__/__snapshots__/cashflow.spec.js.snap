// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cashflow container it should render correctly 1`] = `
<ProjectsTable
  categories={Object {}}
  marketSegments={
    Object {
      "advertising": "Advertising",
      "blockchain_network": "Blockchain Network",
      "data": "Data",
      "digital_identity": "Digital Identity",
      "financial": "Financial",
      "gambling": "Gambling",
      "gaming": "Gaming",
      "legal": "Legal",
      "media": "Media",
      "prediction_market": "Prediction Market",
      "protocol": "Protocol",
      "transportation": "Transportation",
      "unknown": null,
    }
  }
  match={
    Object {
      "path": "/products",
    }
  }
  tableInfo={
    Object {
      "page": 1,
      "pageSize": 32,
      "visibleItems": 32,
    }
  }
/>
`;
